channels/test/liveserver.py

```diff
@@ -0,0 +1,218 @@
+import multiprocessing
+
+import django
+from daphne.server import Server
+from django.core.exceptions import ImproperlyConfigured
+from django.db import connections
+from django.db.utils import load_backend
+from django.test.testcases import TransactionTestCase
+from django.test.utils import modify_settings, override_settings
+from twisted.internet import reactor
+
+from .. import DEFAULT_CHANNEL_LAYER
+from ..asgi import ChannelLayerManager
+from ..staticfiles import StaticFilesConsumer
+from ..worker import Worker, WorkerGroup
+
+# NOTE: We use ChannelLayerManager to prevent layer instance sharing
+# between forked process.  Some layers implementations create
+# connections inside the __init__ method.  After forking child
+# processes can lose the ability to use this connection and typically
+# stuck on some network operation.  To prevent this we use new
+# ChannelLayerManager each time we want to initiate default layer.
+# This gives us guaranty that new layer instance will be created and
+# new connection will be established.
+
+
+class ProcessSetupMixin(object):
```

Comment:
This doesn't need to be a mixin - both places it is used also inherit from `multiprocessing.Process`, so you can just make this inherit from that and have single inheritance.